lib/mix/phoenix/scope.ex

```diff
@@ -0,0 +1,93 @@
+defmodule Mix.Phoenix.Scope do
+  @moduledoc false
+
+  defstruct name: nil,
+            default: false,
+            module: nil,
+            fixture: nil,
+            alias: nil,
+            assign_key: nil,
+            access_path: nil,
+            schema_table: nil,
+            schema_key: nil,
+            schema_type: nil,
+            schema_migration_type: nil
+
+  @doc """
+  Creates a new scope struct.
+  """
+  def new!(name, opts) do
+    scope = struct!(__MODULE__, opts)
+    alias = Module.concat([scope.module |> Module.split() |> List.last()])
+
+    %{
+      scope
+      | name: name,
+        alias: alias,
+        schema_migration_type: scope.schema_migration_type || scope.schema_type
+    }
+  end
+
+  @doc """
+  Returns a `%{name: scope}` map of configured scopes.
+  """
+  def scopes_from_config(otp_app) do
+    scopes = Application.get_env(otp_app, :scopes, [])
+
+    Map.new(scopes, fn {name, opts} -> {name, new!(name, opts)} end)
+  end
+
+  @doc """
+  Returns the default scope.
+  """
+  def default_scope(otp_app) do
+    case Enum.filter(scopes_from_config(otp_app), fn {_, scope} -> scope.default end) do
+      [{_name, scope}] ->
+        scope
+
+      [_ | _] = scopes ->
+        raise """
```

Comment:
We want to use `Mix.raise` in this file. It is specific to Mix and the main difference is that it doesn't include a stacktrace. Those error messages also usually start with uppercase.